mypy_django_plugin/transformers/querysets.py

```diff
@@ -145,10 +175,69 @@ def extract_proper_type_queryset_values_list(ctx: MethodContext, django_context:
     flat = flat or False
     named = named or False
 
-    row_type = get_values_list_row_type(ctx, django_context, model_cls, flat=flat, named=named)
+    is_annotated = is_annotated_model_fullname(model_type.type.fullname)
+    row_type = get_values_list_row_type(
+        ctx, django_context, model_cls, is_annotated=is_annotated, flat=flat, named=named
+    )
     return helpers.reparametrize_instance(ctx.default_return_type, [model_type, row_type])
 
 
+def extract_proper_type_queryset_annotate(ctx: MethodContext, django_context: DjangoContext) -> MypyType:
+    # called on the Instance, returns QuerySet of something
+    assert isinstance(ctx.type, Instance)
+    assert isinstance(ctx.default_return_type, Instance)
+
+    model_type = _extract_model_type_from_queryset(ctx.type)
+    if model_type is None:
+        return AnyType(TypeOfAny.from_omitted_generics)
+
+    model_module_name = model_type.type.module_name
+    type_name = model_type.type.name + ANNOTATED_SUFFIX
+
+    # If already existing annotated type for model exists, reuse it
+    api: CheckerPluginInterface = ctx.api
+    if model_type.type.has_base(ANY_ATTR_ALLOWED_CLASS_FULLNAME):
+        annotated_type = model_type
+    else:
+        annotated_typeinfo = helpers.lookup_fully_qualified_typeinfo(
+            cast(TypeChecker, api), model_module_name + "." + type_name
```

Comment:
If we already use `cast` here and declare `api` as `api: CheckerPluginInterface`, maybe we can change the definition?

Reply:
The definition comes from `ctx` which is `MethodContext` (defined by mypy) as `CheckerPluginInterface`.
I only annotated it locally (redundantly) as `CheckerPluginInterface` since my IDE couldn't figure it out for some reason.
I'll remove the redundant annotation.